### ISSUE TITLE
new_with_options is inconsistent between PNG and JPEG

### ISSUE THREAD

**Issue by kvark** [etemesi254/zune-image#129]
No reason for arguments to be in different order between these two:
```rust
zune_png::PngDecoder::new_with_options(source, options);
zune_jpeg::JpegDecoder::new_with_options(options, source);
```

**Comment by etemesi254** [comment]
Yes, my bad, they were created in different times, 

Ideally there is a breaking change coming since I changed input buffers to be generic, I will make all functions consistent, to be `source`,`options`

**Comment by etemesi254** [comment]
Fixed jpeg to be `source,options` as of 1cc2ffb
